overlord/snapstate/snapstate.go

```diff
@@ -209,6 +209,54 @@ func Update(s *state.State, name, channel string, userID int, flags Flags) (*sta
 	return doInstall(s, snapst.Active, ss)
 }
 
+// Enable sets a snap to the active state
+func Enable(s *state.State, name string) (*state.TaskSet, error) {
```

Comment:
I think we need to check for conflicts with other changes touching this snap, no? like Remove also does

same for Disable